src/words/fast.jl

```diff
@@ -0,0 +1,110 @@
+# TODO: the input + idx combination should probably be replaced by a stream
+
+mutable struct TokenBuffer
+  input::Vector{Char}
+  buffer::Vector{Char}
+  tokens::Vector{String}
+  idx::Int
+end
+
+TokenBuffer(input) = TokenBuffer(input, [], [], 1)
+
+TokenBuffer(input::AbstractString) = TokenBuffer(collect(input))
+
+Base.getindex(ts::TokenBuffer, i = ts.idx) = ts.input[i]
```

Comment:
My style preference is that there are no spaces around the equals sign in kwargs/optional args.

```
Base.getindex(ts::TokenBuffer, i=ts.idx) = ts.input[i]
```

Reply:
I usually avoid the perf annotations unless they benefit performance, do you want them anyway?

Reply:
We can add them later if you want,
I would assume they help,
but we can wait for complains and/or benchmark first.